### DISCUSSION TITLE
Version 11.14.0 is on the way... lessgo

### DISCUSSION THREAD

**Discussion by palexdev** [palexdev/MaterialFX#179] [Show and tell]
Hello everyone, how are you? Hope you're doing well 😉

So, as you can read from the title the new major version is on the way. I don't want to give any ETA since unexpected issues may arise during the development (I really hope not) but I'm working on it **a lot** these days, so I hope to release it very soon.
But let's talk at some of the new features and changes. Some of them are already available on the staging branch and in the latest EA release.

The most important recent changes in my opinion are:
- The introduction of the MFXTitledPane
- The addition of many new icons (some have also been changed to be more appealing)
- The possibility for MFXLabeled controls to show only the graphic node

Oh boys, you should see local changes, here's what' new as of now:
- I added a class to the AnimationUtils that is capable of producing Bezier Easing interpolators, which is 1000 times more intuitive than SPLINE, also thanks to this tool [Cubic Bezier](https://cubic-bezier.com/) (God bless the author)
- MFXScrollBar

Yes, you read that right I decided to make my own scroll bar(done) and scroll pane(will soon start to implement)
The introduction of the scroll bar is not a banal change since I also decided to introduce a new API, the behavior API.
The goal is to allow everyone to customize the behavior of my controls with little to no effort. As how controls are designed now, to change the behavior you have to create a custom skin. The idea is to make MaterialFX controls implement an interface (currently called MFXBehavedControl) which allows you to change the behavior but also to reset it if you need.
The behavior API is not the idea of the century of course. JavaFX has it but...yeah you guessed it, it't private API, what a shame.

So about the scrollbar here's a comparison:
https://user-images.githubusercontent.com/16880178/163735179-6cd1c461-83de-41b7-a192-6bf715c53957.mp4

Well, for now that's all. As usual, stay tuned for updates
Have a nice day!

Trivia
Q1) Why EA versions?
Q2) Why scroll bars and scroll panes?

A1) In one word. Lazyness. Initially I needed the MFXTitledPane control in one of my projects but I didn't want to compile the project, copy the new lib over replacing the one imported from Maven so... I decided to release an EA version lol
A2) The JavaFX scroll pane pissed me off for the last time. So in one of my projects I have this scroll pane and the content changes in size quite often. Did you know that when the scroll bars are not visible in the scroll pane they are not laid out? They are just positioned at [0, 0]. So what do you think happens when the content changes and now it's needed to show the scroll bar? Yep, the content is also re-positioned and it moves up and down...why the fuck...who the heck thought it would be a good idea to make it this way.
Also, as usual I believe the code can be simplified a lot compared to the JavaFX one, improving user experience as a consequence

**Comment by panosru** [comment]
Great work @palexdev ! One question though, is it any particular reason for removing the following icons from the `FontResources`?

```java
DASHBOARD("mfx-dashboard", '\uE924'),
INFO_CIRCLE_FILLED("mfx-info-circle-filled", '\uE946'),
SEARCH("mfx-search", '\uE95E'),
SEARCH_PLUS("mfx-search-plus", '\uE95F'),
```

In your changelog, you mention _"The addition of many new icons (some have also been changed to be more appealing)"_, which is great, but it isn't stated that some icons are removed, which can lead to backwards incompatibility, for instance in my case [I'm using](https://github.com/panosru/CN5004/blob/9e8efc9b7ce98c3daefad21c32862c5eb07e45bd/src/main/java/com/cn5004ap/payroll/common/ModalFactory.java#L165) `INFO_CIRCLE_FILLED`, therefore I'm getting an exception after upgrading to v11.14, which I did because of `MFXTitledPane` addition.

**Addition:**
For `MFXTitledPane`, so far what I think could be improved is to hide the control arrow when the _collapsible_ property is set to _false_.

**Reply by palexdev** [reply]
Hello @panosru sorry for the late response, been quite busy these last months
So, I made this change quite some time ago so I don't really remember what I changed exactly but I think the cases are two:
1. The icon was removed because it was too unappealing
2. I just changed their name since the resource changed too. For example the info circle filled now is just INFO_CIRCLE, and the unfilled one is INFO_CIRCLE_LIGHT. The search icons are now magnifying glasses
